pkg/provisioner/crdprovisioner.go

```diff
@@ -155,29 +155,34 @@ func (c *CrdProvisioner) CreateVolume(
 			// The volume creation was successful previously,
 			// Returning the response object from the status
 			return azVolumeInstance.Status.Detail.ResponseObject, nil
-		} else if azVolumeInstance.Status.Error != nil {
-			updateFunc := func(obj interface{}) error {
-				updateInstance := obj.(*v1alpha1.AzVolume)
-				updateInstance.Status.Error = nil
-				updateInstance.Status.State = v1alpha1.VolumeOperationPending
-
-				if !isAzVolumeSpecSameAsRequestParams(updateInstance, maxMountReplicaCount, capacityRange, parameters, secrets, volumeContentSource, accessibilityReq) {
-					// Updating the spec fields to keep it up to date with the request
-					updateInstance.Spec.MaxMountReplicaCount = maxMountReplicaCount
-					updateInstance.Spec.CapacityRange = capacityRange
-					updateInstance.Spec.Parameters = parameters
-					updateInstance.Spec.Secrets = secrets
-					updateInstance.Spec.ContentVolumeSource = volumeContentSource
-					updateInstance.Spec.AccessibilityRequirements = accessibilityReq
-				}
+		}
+		// return if volume creation is already in process to prevent duplicate request
+		if azVolumeInstance.Status.State == v1alpha1.VolumeCreating {
+			return nil, status.Errorf(codes.Internal, "creation still in process for volume (%s)", volumeName)
```

Comment:
Generally, `codes.Aborted` is returned for duplicate operations. See [Container Storage Interface / Spec / Error Scheme](https://github.com/container-storage-interface/spec/blob/master/spec.md#error-scheme)